cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -175,9 +189,17 @@ public ClassLoader getArtifactClassLoader(ArtifactId artifactId) throws IOExcept
       Field field = Fields.findField(pluginType.getType(), configFieldName);
       TypeToken<?> configFieldType = pluginType.resolveType(field.getGenericType());
       Object config = instantiatorFactory.get(configFieldType).create();
-      Reflections.visit(config, configFieldType.getType(),
-                        new ConfigFieldSetter(pluginClass, plugin.getArtifactId(), plugin.getProperties(),
-                                              getFieldsWithMacro(plugin)));
+
+      // perform macro substitution if an evaluator is provided
+      if (macroEvaluator == null) {
+        Reflections.visit(config, configFieldType.getType(),
+                          new ConfigFieldSetter(pluginClass, plugin.getArtifactId(), plugin.getProperties(),
+                                                getFieldsWithMacro(plugin)));
+      } else {
+        Reflections.visit(config, configFieldType.getType(),
+                          new ConfigFieldSetter(pluginClass, plugin.getArtifactId(),
+                                                substituteMacroFields(plugin, macroEvaluator), new HashSet<String>()));
```

Comment:
Hm I guess the question is how we expect people to use the containsMacro method. Is there any reason somebody would call it at runtime and use it for something useful? It seems like it would never be useful if we always return false, but it may be useful if we return whether macros were evaluated or not.